test/core/tsi/ssl_transport_security_test.cc

```diff
@@ -558,6 +581,42 @@ void ssl_tsi_test_do_round_trip_odd_buffer_size() {
   }
 }
 
+void ssl_tsi_test_do_handshake_session_cache() {
+  tsi_ssl_session_cache* session_cache = tsi_ssl_session_cache_create_lru(16);
+
+  char stek[48];
+
+  auto do_handshake = [&stek, session_cache](bool session_reused) {
+    tsi_test_fixture* fixture = ssl_tsi_test_fixture_create();
+    ssl_tsi_test_fixture* ssl_fixture =
+        reinterpret_cast<ssl_tsi_test_fixture*>(fixture);
+    ssl_fixture->server_name_indication =
+        const_cast<char*>("waterzooi.test.google.be");
+    ssl_fixture->stek = stek;
+    ssl_fixture->stek_size = 48;
+    ssl_fixture->session_cache = session_cache;
+    ssl_fixture->session_reused = session_reused;
+    tsi_test_do_round_trip(&ssl_fixture->base);
+    tsi_test_fixture_destroy(fixture);
+  };
+
```

Comment:
Please add a test where server does not provide session ticket key, thus session ticket key was generated by SSL. In this case, still observe session reused.

Reply:
At this moment tsi test framework re-creates new handshaker factory and handshaker for every round trip and doesn't allow re-usage of factory between handshakes. This means that new session ticket is regenerated every time effectively making ticket caching useless.

Should I refactor tsi test framework in this cl to allow re-usage of the same handshaker factory between different handshakes?

Reply:
Ah. I see. No need to refactor tsi test framework. We don't have to test this then.